technic/config.lua

```diff
@@ -1,28 +1,74 @@
-technic.config = technic.config or Settings(minetest.get_worldpath().."/technic.conf")
-
-local conf_table = technic.config:to_table()
+local config_file = minetest.get_worldpath() .. "/technic.conf"
 
 local defaults = {
+	-- Power tools enabled
+	enable_cans = "true",
+	enable_chainsaw = "true",
+	enable_flashlight = "false",
 	enable_mining_drill = "true",
 	enable_mining_laser = "false",
-	enable_flashlight = "false",
+	enable_multimeter = "true",
+	enable_prospector = "true",
+	enable_sonic_screwdriver = "true",
+	enable_tree_tap = "true",
+	enable_vacuum = "true",
+
+	-- Power tool options
+	multimeter_remote_start_ttl = "300",
+
+	-- Machine options
 	enable_wind_mill = "false",
 	enable_frames = "false",
+	enable_nuclear_reactor_digiline_selfdestruct = "false",
+	quarry_dig_above_nodes = "3",
+	quarry_max_depth = "100",
+	quarry_time_limit = "5000",
+
+	-- Power network and general options
+	switch_off_delay_seconds = "1800",
+	network_overload_reset_time = "20",
+	admin_priv = "basic_privs",
 	enable_corium_griefing = "true",
 	enable_radiation_protection = "true",
 	enable_radiation_throttling = "false",
 	enable_entity_radiation_damage = "true",
 	enable_longterm_radiation_damage = "true",
-	enable_nuclear_reactor_digiline_selfdestruct = "false",
-	switch_off_delay_seconds = "1800",
-	network_overload_reset_time = "20",
-	admin_priv = "basic_privs",
-	quarry_dig_above_nodes = "3",
-	quarry_max_depth = "100",
-	quarry_time_limit = "5000",
 	--constant_digit_count = nil,
 }
 
+--
+-- Create technic.config settings object and
+-- initialize configuration with default values.
+--
+
+local config_obj = Settings(config_file)
+local Config = {}
+Config.__index = Config
+
+function Config:get_int(...)
+	local args = {...}
+	local value = tonumber(self:get(args[1]))
+	if not value then
+		if args[2] then
+			return args[2]
+		end
+		error("Invalid configuration value for key " .. args[1] .. " in " .. config_file .. ". Number expected.")
+	end
+	return value
+end
+
+local config_obj_mt = getmetatable(config_obj)
+for key,value in pairs(config_obj_mt) do
+	if type(value) == "function" then
+		Config[key] = function(self, ...) return value(config_obj, ...) end
+	else
+		config_obj_mt[key] = value
+	end
+end
+setmetatable(Config, config_obj_mt)
+technic.config = setmetatable({}, Config)
```

Comment:
Not sure if this is a bit over-engineered. Would it be better to just define `technic.config` object to be table with functions that either forward calls to underlying Settings object (`get`, `write`, `to_table`, etc.) or implement call (`get_int`)?
I feel this is over-engineered as in overly complex solution to very simple problem...

Reply:
uhm, i haven't checked every setting related code, but why don't we use hardcoded setting-keys (`technic.*`) and `minetest.settings:get_*` everywhere?

Reply:
Well, `minetest.settings` is `Settings` object and therefore `:get_*` does not include `get_int` which was one attempted improvement here.
Everything or at least almost everything is in `worldpath/technic.conf` already.

`technic.config` is `Settings` object and therefore has exact same API that `minetest.settings` offers.

Some maybe about 20 new settings would be added on top of current about 20 options, none of those are currently prefixed in any way and I think count is high enough and options specific enough to warrant using separate configuration file.

Reply:
Updated technic.config forwarders to be significantly simpler and easier to understand.